src/lib.rs

```diff
@@ -3884,6 +3887,17 @@ mod tests {
         )
     }
 
+    #[test]
+    fn separated_by() {
+        use crate::{error::Simple, extra};
+
+        let parser = just::<_, &str, extra::Err<Simple<_>>>("a")
+            .or_not()
+            .separated_by(just("b"));
+
+        assert_eq!(parser.parse("bba").into_result(), Ok(()));
+    }
+
```

Comment:
Given that you've changed the behaviour of the debug check, could you add a simple `#[should_panic]` test to ensure we do still hit it in debug mode? (it'll need a cfg(debug_assertions), ofc)

```rs
#[test]
    #[cfg(debug_assertions)]
    #[should_panic]
    fn debug_error_hit() {
        use crate::{error::Simple, extra};

        let parser = just::<_, &str, extra::Err<Simple<_>>>("a")
            .or_not()
            .repeated();

        assert_eq!(parser.parse("").into_result(), Ok(()));
    }
```